### PR TITLE
Resizable tof spectrum viewer

### LINKED ISSUE #2025: Resizable ToF Spectrum plot in Spectrum Viewer
### Desired Behaviour

The IMAT Scientists have requested the ability to resize the Spectrum Viewer such that the ToF spectrum plot is easier to view. 

### Current Behaviour

Currently, the plot can be zoomed into and dragged around but the plot itself cannot be made any larger which can make it difficult to see small details, especially when multiple overlapping ROIs are active.

### PR DESCRIPTION
**Pull request by MikeSullivan7**
### Issue

Closes #2025.

### Description

In the `SpectrumWidget` for the Spectrum Viewer, the image view and the spectrum plot have been separated into two separate widgets as this allows us to use a QSplitter to resize them in a comfortable way. The splitter is configured so that initially the image takes up 70% of the window and the spectrum plot takes 30%, but this can be easily configured. The `SpectrumWidget` class now creates instances of the `SpectrumPlotWidget` and `SpectrumProjectionWidget`. This refactoring required the unit tests to be modified.

image

image

### Testing 

Check `make check` still passes properly.

### Acceptance Criteria 

Open some data in the Spectrum Viewer and see if the ToF plot can be easily resized and viewed. Make sure that the rest of the Spectrum Viewer still works as intended.

### Documentation

Will be added to release notes.

### CODE REVIEW

File: mantidimaging/gui/windows/spectrum_viewer/view.py [hunk 1]
```diff
@@ -41,6 +41,8 @@ class SpectrumViewerWindowView(BaseMainWindowView):
     bin_size_spinBox: QSpinBox
     bin_step_spinBox: QSpinBox
 
+    spectrumWidget: SpectrumWidget
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/spectrum_viewer/view.py:44]: Can the local name be in snake_case
`spectrum_widget: SpectrumWidget`

Comment (MikeSullivan7) [reply mantidimaging/gui/windows/spectrum_viewer/view.py]: done!

**Comment by samtygier-stfc** [comment]
Looks good. Just needs a release note

**Review by samtygier-stfc — APPROVED** [review]
Works well. Thanks

**Pull request merged**